File: PYTHON/BAEKJOON/2750_수_정렬하기(버블,선택,삽입)/2750_SelectionSort.py
N = int(input())
arr = [int(input()) for _ in range(N)]

for i in range(N - 1, 0, -1):  # N-1번째 인덱스부터 ~ 1 까지
    biggest = [i, arr[i]]  # 초기에 가장 큰 수 및 인덱스
    for j in range(0, i):  # 0 ~ i 번째 인덱스 수까지 돌면서
        # 최댓값을 찾는다.
        if biggest[1] < arr[j]:
            biggest = [j, arr[j]]

    arr[i], arr[biggest[0]] = arr[biggest[0]], arr[i]

for v in arr:
    print(v)

File: BAEKJOON/2750_수_정렬하기(버블,선택,삽입)/2750_SelectionSort.js
const fs = require('fs');
const fileStream = fs.createReadStream('input.txt');
const readline = require('readline');

const rl = readline.createInterface({
  input: fileStream,
  output: process.stdout,
})

let N;
let input = []
let lineIdx = 0

function selectionSort(arr, N) {
  for (let i = N - 1; i > 0; i--) {
    let biggest = arr[i];
    let biggestIdx = i;
    for (let j = 0; j < i; j++) {
      const compare = arr[j]
      if (compare > biggest) {
        biggest = compare;
        biggestIdx = j;
      }
    }

    const temp = arr[i]
    arr[i] = arr[biggestIdx];
    arr[biggestIdx] = temp;
  }
  return arr;
}

rl.on("line", (line) => {
  if (lineIdx === 0) {
    N = Number(line)
    lineIdx++;
  } else {
    input.push(Number(line))
  }
})
  .on("close", () => {
    // 로직
    selectionSort(input, N).forEach(v => console.log(v));
    process.exit()
  })